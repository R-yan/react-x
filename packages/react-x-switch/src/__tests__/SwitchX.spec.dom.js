// Modules
import React from 'react';
import SwitchX from '../SwitchX';


/* eslint-disable no-undef */
describe('SwitchX.dom', () => {
  context('with default props', () => {
    forEach({isOn: [true, false]}, () => {
      itRenders(() => <SwitchX isOn={isOn} />);
    });
  });
});
