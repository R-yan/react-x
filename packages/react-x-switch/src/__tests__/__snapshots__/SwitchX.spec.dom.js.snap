// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchX.dom with default props renders snapshot 1`] = `
<div
  className="react-ios-switch-Switch-switch"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(229, 229, 229)",
    }
  }
>
  <div
    className="react-ios-switch-Switch-offState"
    style={
      Object {
        "WebkitTransform": "scale(1)",
        "backgroundColor": "white",
        "msTransform": "scale(1)",
        "transform": "scale(1)",
      }
    }
  />
  <div
    className="react-ios-switch-Switch-handle"
    onClick={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "WebkitTransform": "translateX(0px)",
        "backgroundColor": "white",
        "msTransform": "translateX(0px)",
        "transform": "translateX(0px)",
      }
    }
  />
  <input
    checked={false}
    className="react-ios-switch-Switch-input"
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`SwitchX.dom with isDisabled set to false renders snapshot 1`] = `
<div
  className="react-ios-switch-Switch-switch"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(229, 229, 229)",
    }
  }
>
  <div
    className="react-ios-switch-Switch-offState"
    style={
      Object {
        "WebkitTransform": "scale(1)",
        "backgroundColor": "white",
        "msTransform": "scale(1)",
        "transform": "scale(1)",
      }
    }
  />
  <div
    className="react-ios-switch-Switch-handle"
    onClick={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "WebkitTransform": "translateX(0px)",
        "backgroundColor": "white",
        "msTransform": "translateX(0px)",
        "transform": "translateX(0px)",
      }
    }
  />
  <input
    checked={false}
    className="react-ios-switch-Switch-input"
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`SwitchX.dom with isDisabled set to true renders snapshot 1`] = `
<div
  className="react-ios-switch-Switch-switch"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(229, 229, 229)",
    }
  }
>
  <div
    className="react-ios-switch-Switch-offState"
    style={
      Object {
        "WebkitTransform": "scale(1)",
        "backgroundColor": "white",
        "msTransform": "scale(1)",
        "transform": "scale(1)",
      }
    }
  />
  <div
    className="react-ios-switch-Switch-handle"
    onClick={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "WebkitTransform": "translateX(0px)",
        "backgroundColor": "white",
        "msTransform": "translateX(0px)",
        "transform": "translateX(0px)",
      }
    }
  />
  <input
    checked={false}
    className="react-ios-switch-Switch-input"
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`SwitchX.dom with isOn set to false renders snapshot 1`] = `
<div
  className="react-ios-switch-Switch-switch"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(229, 229, 229)",
    }
  }
>
  <div
    className="react-ios-switch-Switch-offState"
    style={
      Object {
        "WebkitTransform": "scale(1)",
        "backgroundColor": "white",
        "msTransform": "scale(1)",
        "transform": "scale(1)",
      }
    }
  />
  <div
    className="react-ios-switch-Switch-handle"
    onClick={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "WebkitTransform": "translateX(0px)",
        "backgroundColor": "white",
        "msTransform": "translateX(0px)",
        "transform": "translateX(0px)",
      }
    }
  />
  <input
    checked={false}
    className="react-ios-switch-Switch-input"
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;

exports[`SwitchX.dom with isOn set to true renders snapshot 1`] = `
<div
  className="react-ios-switch-Switch-switch"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(0, 0, 0)",
    }
  }
>
  <div
    className="react-ios-switch-Switch-offState"
    style={
      Object {
        "WebkitTransform": "scale(0)",
        "backgroundColor": "white",
        "msTransform": "scale(0)",
        "transform": "scale(0)",
      }
    }
  />
  <div
    className="react-ios-switch-Switch-handle"
    onClick={[Function]}
    onMouseDown={[Function]}
    style={
      Object {
        "WebkitTransform": "translateX(20px)",
        "backgroundColor": "white",
        "msTransform": "translateX(20px)",
        "transform": "translateX(20px)",
      }
    }
  />
  <input
    checked={true}
    className="react-ios-switch-Switch-input"
    disabled={false}
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
